plugins/modules/elastic_pkgpolicy.py

```diff
@@ -77,29 +84,50 @@ def main():
       results['changed'] = False
       module.exit_json(**results)
     
-    if module.params.get('integration_name'):
-      integration_object = kibana.check_integration(integration_name)
+    if module.params.get('integration_title'):
+      integration_object = kibana.check_integration(integration_title)
     else:
       results['integration_status'] = "No Integration Name provided to get the integration object"
       results['changed'] = False
       module.exit_json(**results)
     
-    if not integration_object:
-      results['integration_status'] = 'Integration name is not a valid'
+    if ( integration_name and integration_ver and integration_name) and not integration_object:
+      results['integration_status'] = "No integration found, but Integration Name, Version, and Title found"
+      integration_object = {
+        'name': integration_name,
+        'title': integration_title,
+        'version': integration_ver
+      }
+    elif not integration_object and not ( integration_name and integration_ver and integration_name):
+      results['integration_status'] = 'Integration Title is not valid and integration name and integration version are not found'
       results['changed'] = False
-      module.exit_json(**results)
+      module.exit_json(**results) 
     
     if state == "present":
-      pkg_policy_object = kibana.get_pkg_policy(integration_name,agent_policy_id)
+      pkg_policy_object = kibana.get_pkg_policy(integration_title,agent_policy_id)
       if pkg_policy_object:
         results['pkg_policy_status'] = "Integration Package found, No package created"
         results['changed'] = False
       else:
-        pkg_policy_object = kibana.create_pkg_policy(pkg_policy_name, pkg_policy_desc, agent_policy_id, integration_object['response'], namespace)
+        pkg_policy_object = kibana.create_pkg_policy(pkg_policy_name, pkg_policy_desc, agent_policy_id, integration_object, namespace)
```

Comment:
If `integration_object` is set on line 88, we wouldn't have grabbed the value from the `response` key yet?

Reply:
this was updated where appropriate